broni/shapes/primitives.py

```diff
@@ -13,17 +13,9 @@ def __init__(self, x: float, y: float, z: float, r: float):
         self.radius = r
 
     def intersect(self, trajectory: Trajectory):
-        dist = np.linalg.norm(self.center - trajectory.data, axis=1)
+        dist = np.linalg.norm(self.center - trajectory.cartesian()[:, 0:3], axis=1)
```

Comment:
not sure the ```[:, 0:3]``` is needed here

Reply:
You're right, not needed anymore because sphericial will be another member.